src/main/java/com/emt/service/MatchService.java

```diff
@@ -65,4 +69,36 @@ public BigDecimal calculateProbability(BigDecimal rating1, BigDecimal rating2) {
 
     return ONE.divide(divisor, 2, HALF_UP);
   }
+
+  public void cancelMatch(Long matchId) {
+    Match matchToCancel =
+        matchRepository.findById(matchId).orElseThrow(() -> new MatchNotFoundException(matchId));
+
+    Player winner = matchToCancel.getWinner();
+    Player loser = matchToCancel.getLoser();
+    BigDecimal winnerRatingChange = matchToCancel.getWinnerRatingChange();
+
+    winner.setEloRating(winner.getEloRating().subtract(winnerRatingChange));
+    loser.setEloRating(loser.getEloRating().add(winnerRatingChange));
+
+    playerService.saveWinnerAndLoser(winner, loser);
+
+    List<Match> subsequentMatches =
+        matchRepository.findMatchesByPlayersAfter(
+            matchToCancel.getCreatedAt(), winner.getPlayerId(), loser.getPlayerId());
+
+    recalculateEloRatingsForSubsequentMatches(subsequentMatches);
+
+    matchToCancel.setCancelled(true);
+    matchRepository.deleteById(matchId);
+  }
```

Comment:
This method performs multiple interdependent operations that must either all succeed or all fail together, so consider how to ensure consistency